src/Application/App.hpp

```diff
@@ -327,6 +331,11 @@ class App : public KPController, public TaskObserver {
         }
 
         for (auto v : task.valves) {
+            if(v > vm.valves.size()){
+                KPStringBuilder<100> error("Valve ", v, " is not in device configuration");
```

Comment:
Check if there's a better name than Valve (make a backlog task if this ends up blocking)